libsignal-protocol/src/session_cipher.rs

```diff
@@ -66,6 +66,44 @@ impl SessionCipher {
             })
         }
     }
+
+    /// Decrypt a pre key message
+    pub fn decrypt_pre_key_message(
+        &self,
+        message: &PreKeySignalMessage,
+    ) -> Result<Buffer, Error> {
+        unsafe {
+            let mut buffer = ptr::null_mut();
+            sys::session_cipher_decrypt_pre_key_signal_message(
+                self.raw,
+                message.raw.as_ptr(),
+                ptr::null_mut(),
```

Comment:
Is this something we should do now? What use case do you see for this?

Reply:
I was thinking it might be usable for things like logging, middleware, or progress monitoring.

I doubt we'd need it now, but it's nice to know this sort of thing exists for later.